project/Boilerplate.scala

```diff
@@ -240,7 +244,38 @@ object Boilerplate {
          |trait ParallelArityFunctions {
         -  /** @group ParMapArity */
         -  def parMap$arity[M[_], F[_], ${`A..N`}, Z]($fparams)(f: (${`A..N`}) => Z)(implicit p: NonEmptyParallel[M, F]): M[Z] =
-        -    p.flatMap.map($nestedProducts) { case ${`nested (a..n)`} => f(${`a..n`}) }
+        -    p.flatMap.map(${nestedExpansion.products}) { case ${nestedExpansion.`(a..n)`} => f(${`a..n`}) }
+         |}
+      """
+    }
+  }
+
+  object GenParallelArityFunctions2 extends Template {
+    def filename(root: File) = root / "cats" / "ParallelArityFunctions2.scala"
+    override def range = 2 to maxArity
+    def content(tv: TemplateVals) = {
+      import tv._
+
+      val tpes = synTypes map { tpe => s"M[$tpe]" }
+      val fargs = (0 until arity) map { "m" + _ }
+      val fparams = (fargs zip tpes) map { case (v,t) => s"$v:$t"} mkString ", "
+      val fargsS = fargs mkString ", "
+      val nestedExpansion = ParallelNestedExpansions(arity)
+
+      block"""
+         |package cats
+         |
+         |/**
+         | * @groupprio Ungrouped 0
+         | *
+         | * @groupname ParTupleArity parTuple arity
+         | * @groupdesc ParTupleArity Higher-arity parTuple methods
+         | * @groupprio ParTupleArity 999
+         | */
+         |trait ParallelArityFunctions2 {
```

Comment:
If we make this one an abstract class then we can add more methods in the future without breaking bin compat.

Reply:
sure, changed. Thanks